### ISSUE TITLE
Option to fix vertical cross section in 3D viewer GUI

### ISSUE THREAD

**Issue by TomFarley** [euratom-software/calcam#53]
It would be really nice to be able to select a check box in the 3D viewer GUI to force the cross-section cut to always be in a vertical plane (perhaps just through the origin if the cursor is more difficult) rather than the default of the plane whose normal is the viewing direction. 

I don't know how difficult this would be to implement, but it would enable some really nice figures where you have a cross section of the machine, but could have a tilted view eg looking down at the divertor tiles you are interested in.

**Comment by ssilburn** [comment]
Yep agreed this would be really good! But it will need a complete change in how the cross-sectioning is implemented. At the moment I "cheat" and do the cross-sectioning by setting the rendering clipping distance of the camera to achieve a cross-section, which makes the cross-section plane necesserily normal to the view direction. To implement what you suggest here I'd have to change to doing boolean operations on the 3D geometry. Should be perfectly possible to do and as you say it would allow some really nice figure making, but also would probably be a lot slower than the current implementation.
